cloud/scope/cluster.go

```diff
@@ -283,3 +299,31 @@ func (s *ClusterScope) SetFailureDomain(id string, spec clusterv1.FailureDomainS
 	}
 	s.AzureCluster.Status.FailureDomains[id] = spec
 }
+
+func (s *ClusterScope) ControlPlaneIngressRules() infrav1.IngressRules {
+	if s.ControlPlaneSubnet().SecurityGroup.IngressRules != nil {
+		return s.ControlPlaneSubnet().SecurityGroup.IngressRules
+	}
+	return infrav1.IngressRules{
+		&infrav1.IngressRule{
+			Name:             "allow_ssh",
+			Description:      "Allow SSH",
+			Priority:         100,
```

Comment:
What would you think about having a higher base `Priority`. When starting with 100, there isn't much room to make rules ahead of this. Perhaps, 2000, or some other arbitrary, configurable base.

This feature probably shouldn't be in this refactor, just remembered seeing this change.

Reply:
Not a bad idea. Not sure what the implication would be for reconciling pre-existing clusters with the old priority though so will look into this separately. These rules can be overridden by the user with a set of custom rules so there is at least that workaround in the meantime.